src/algorithms/signal_source/gnuradio_blocks/fifo_reader.cc

```diff
@@ -0,0 +1,111 @@
+/*!
+ * \file fifo_reader.cc
+ *
+ * \brief Implementation of the class to retrieve samples from an existing Unix FIFO
+ * \author Malte Lenhart, 2021. malte.lenhart(at)mailbox.org
+ *
+ * -----------------------------------------------------------------------------
+ *
+ * GNSS-SDR is a Global Navigation Satellite System software-defined receiver.
+ * This file is part of GNSS-SDR.
+ *
+ * Copyright (C) 2010-2020  (see AUTHORS file for a list of contributors)
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ * -----------------------------------------------------------------------------
+ */
+
+#include "fifo_reader.h"
+#include <glog/logging.h>
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <string>
+
+// initial construction; pass to private constructor
+FifoReader::sptr FifoReader::make(const std::string &file_name, const std::string &sample_type)
+{
+    return gnuradio::get_initial_sptr(new FifoReader(file_name, sample_type));
+}
+
+// private constructor called by ::make
+FifoReader::FifoReader(const std::string &file_name, const std::string &sample_type)
+    : gr::sync_block("fifo_reader",
+          gr::io_signature::make(0, 0, 0),                    // no input
+          gr::io_signature::make(1, 1, sizeof(gr_complex))),  // <+MIN_OUT+>, <+MAX_OUT+>, sizeof(<+OTYPE+>)
+      file_name_(file_name),
+      sample_type_(sample_type)
+{
+    std::cout << "Starting FifoReader\n";
```

Comment:
be sure to flush your std::cout stream ... don't assume it is a terminal and line-buffered.

Reply:
I'll change it to glog as it is the only std::cout in this file. thanks!
(btw, is there a preference to use LOG() or DLOG() ? one is just defined as alias for the other..)